src/pnm/decoder.rs

```diff
@@ -616,10 +616,7 @@ impl<R: Read> PnmDecoder<R> {
 fn read_separated_ascii<T: FromStr<Err = ParseIntError>>(reader: &mut dyn Read) -> ImageResult<T>
     where T::Err: Display
 {
-    let is_separator = |v: &u8| match *v {
-        b'\t' | b'\n' | b'\x0b' | b'\x0c' | b'\r' | b' ' => true,
-        _ => false,
-    };
+    let is_separator = |v: &u8| matches! { *v, b'\t' | b'\n' | b'\x0b' | b'\x0c' | b'\r' | b' '};
```

Comment:
Very nice, but requires rust 1.42.0 (of course, this is about all uses of `matches!`).

Reply:
Thank you! I've reverted this.